src/compiler/checker.ts

```diff
@@ -2194,10 +2194,17 @@ namespace ts {
                 }
             }
 
-            function buildDisplayForParametersAndDelimiters(parameters: Symbol[], writer: SymbolWriter, enclosingDeclaration?: Node, flags?: TypeFormatFlags, symbolStack?: Symbol[]) {
+            function buildDisplayForParametersAndDelimiters(thisType: Type, parameters: Symbol[], writer: SymbolWriter, enclosingDeclaration?: Node, flags?: TypeFormatFlags, symbolStack?: Symbol[]) {
                 writePunctuation(writer, SyntaxKind.OpenParenToken);
+                const useThisType = thisType && thisType.symbol;

```

Comment:
`voidType` doesn't have a symbol, which avoids displaying (this: void, ...) for every single function, especially those that weren't annotated in the first place.